src/jdataview.d.ts

```diff
@@ -0,0 +1,133 @@
+// We need to add all of these types as getXXX setXXX
+// and writeXXX methods on the jDataView class.
+// Various type acrobatics are needed to make this work.
+
+type RegularTypes = {
+	Float64: number;
+	Float32: number;
+	BigInt64: bigint;
+	BigUint64: bigint;
+	Int32: number;
+	Uint32: number;
+	Int16: number;
+	Uint16: number;
+	Int8: number;
+	Uint8: number;
+	Int64: number;
+	Uint64: number;
+	Bytes: ArrayLike<number>;
+};
+
+// These are the actual getXXX, setXXX, and writeXXX function definitions
+type GetTypeFunction<T extends keyof RegularTypes> = (
+	byteOffset: number | undefined,
+	littleEndian?: boolean | undefined
+) => RegularTypes[T];
+type SetTypeFunction<T extends keyof RegularTypes> = (
+	byteOffset: number | undefined,
+	value: RegularTypes[T],
+	littleEndian?: boolean | undefined
+) => void;
+type WriteTypeFunction<T extends keyof RegularTypes> = (
+	value: RegularTypes[T],
+	arg3?: boolean | undefined
+) => void;
+
+// First off, each object type can only have one mapped type, so we
+// have to do each in a separate object and then merge them all.
+type GetSetWrite = {
+	[T in keyof RegularTypes as `get${T}`]: GetTypeFunction<T>;
+} & { [T in keyof RegularTypes as `set${T}`]: SetTypeFunction<T> } & {
+	[T in keyof RegularTypes as `write${T}`]: WriteTypeFunction<T>;
+};
+
+// And then we need to use an interface to 'extend' the type, because
+// interfaces can extend type but classes can't.
+// Then, because the interface has the same name as the class, their
+// types are automatically merged.
+// Also we need to omit `getBytes` because just signature for the getter
+// is different. The setter and writer are the same.
+interface jDataView extends Omit<GetSetWrite, "getBytes"> {}
+
+// Regular, non-cursed typescript from here on
+
+type TypedArray =
+	| Int8Array
+	| Uint8Array
+	| Uint8ClampedArray
+	| Int16Array
+	| Uint16Array
+	| Int32Array
+	| Uint32Array
+	| Float32Array
+	| Float64Array
+	| BigInt64Array
+	| BigUint64Array;
+
+type Bufferish = string | number | ArrayBuffer | TypedArray | ArrayLike<number>;
+
+declare class jDataView {
```

Comment:
```suggestion
declare class jDataView implements DataView {
```

might be helpful as a self-check that we are still a drop-in replacement. E.g. if that interface above declared some incompatible signature, TS would complain here.

Right now it seems to complain only about missing `Symbol.toStringTag` - we should probably provide a custom implementation there.